client/src/components/Login/index.js

```diff
@@ -1,32 +1,105 @@
-import React from "react";
+import React, { Component } from "react";
 import KS_logo from "./KS_logo.png";
 import Mercy from "./Mercy .png";
 import Button from "@material-ui/core/Button";
 import TextField from "@material-ui/core/TextField";
-const Login = () => {
-  return (
-    <div className="header">
-      <div>
-        <img className="logo" alt="logo" src={KS_logo} />
-      </div>
-      <div className="title">KamSafer </div>
-      <div className="form">
-        <form>
-          <label>
-            Username:
-            <TextField type="text" name="username" />
-          </label>
-          <label>
-            Password:
-            <TextField type="Password" name="password" />
-          </label>
-          <Button type="submit" value="Log in" />
-        </form>
-      </div>
-      <div>
-        <img className="logos" alt="logo" src={Mercy} />
+import { withStyles } from "@material-ui/core/styles";
+import PropTypes from "prop-types";
+const styles = {
+  button: {
+    backgroundColor: "red"
+  }
+};
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: "",
+      password: "",
+      failed: false
+    };
+  }
+  handleInputChange = ({ target: { name, value } }) => {
+    this.setState({
+      [name]: value
+    });
+  };
+
+  handleSubmit = event => {
+    const { history } = this.props;
+    const username = this.state.username;
+    const password = this.state.password;
+    event.preventDefault();
+    const data = JSON.stringify({ username, password });
+    fetch("/login", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: data
+    })
+      .then(response => {
+        return response.json();
+      })
+      .then(result => {
+        if (result.logged) {
+          history.push("/cars");
+        } else {
+          this.setState({
+            failed: true
+          });
+        }
+      })
+      .catch(error => {
+        console.log(error);
+      });
```

Comment:
There should be an error message something like `An error has occurred please try again` instead of `console.log`

Reply:
Also the function request could be wrapped in a function that is in a different folder like `util`